src/late_multicellular_stage/SCALIS.cs

```diff
@@ -0,0 +1,272 @@
+﻿using System;
+using System.Collections.Generic;
+using Godot;
+
+public class Scalis : IMeshGeneratingFunction
+{
+    public float CutoffPointMultiplier = 1.5f;
+
+    public float InnerCutoffPointMultiplier = 2.0f;
+
+    public bool Cutoff = true;
+
+    /// <summary>
+    ///   Every two consecutive points form a bone
+    /// </summary>
+    public Metaball[]? Points;
+
+    private static int[] coefficients = { 1, 2, 1 };
+
+    private float surfaceValue = 1.0f;
+
+    public float SurfaceValue
+    {
+        get => surfaceValue;
+        set => surfaceValue = value;
+    }
+
+    public void FindBones(IReadOnlyCollection<MulticellularMetaball> layout)
```

Comment:
What if the constructor was already made to require to be given the layout and it would calculate the Points? That way it can be read-only.

Also I think the array could be created directly without an intermediate list as it should be the case that there's exactly one root metaball, so the list length is already known based on the layout item count. This will save on allocating an extra list.